### PR TITLE
Gangs Rework - Less Greytide Stationwide, More Territory Control

### PR DESCRIPTION
**Pull request by eeSPee**
## About The Pull Request

The ideas of the gang rework:
* Removed dominators, instead the objective of the gang is to control most territories by the end of the shift
* Reworked brainwashing/gang implants; instead of removing mindshields, they will be used for reviving your own or converting enemy gangmembers to your alignment
* A problem with gangs was that they were really hard to tell apart from regulars; now there's more incentive for them to wear their gangwear/costumes; gangsters will not be able to generate income, take over territories or use gangtools unless costumed
* Another problem with gangs was how easy it was for them to obtain powerful gear. The equipment they gain will be more utility and less combat

## Why It's Good For The Game

Gang was an interesting game mode, but gangs were too powerful and it was really easy for them to overwhelm command and descend into total chaos.

Everyone likes GTA:SA's gang domination, but gangs is really improperly done.

Dominator and implant removal? Really?

## Tbd
* [x] Gang Outfits
* [x] Revenue Revise
* [x] Remove Domm
* [x] Implant Rework
* [x] Economy Revise
* [x] Victory Conditions
* [x] Flavor Text
* [x] Testing

## Changelog
:cl:
add: Gangs Rework
/:cl:

### CODE REVIEW

**Comment by heepox** [comment]
Solid idea. Maybe take a pointer from goon with its gang locker as a way to make more gang points for gear. Inserting drugs into the locker gives points, but controlling territory gives a more steady income. So if a gang controls botany or chemistry they will be doing drugs? Gang with most points wins instead of territory as that is gonna end with people marking and walling off places with R walls to hide gang signs.

Also inserting money into the locker or weaponry would also give points (control cargo and budgets and you have more power)

**Comment by eeSPee** [comment]
> Solid idea. Maybe take a pointer from goon with its gang locker as a way to make more gang points for gear. Inserting drugs into the locker gives points, but controlling territory gives a more steady income. So if a gang controls botany or chemistry they will be doing drugs? Gang with most points wins instead of territory as that is gonna end with people marking and walling off places with R walls to hide gang signs.

Hmm. I like the idea of winning based on influence.

But I don't like the idea of selling drugs for points. I was gonna have them feed money to the gang tool to rack up points and have to sell stuff for money.

Botany already has a lot of stuff that can help gangs, such as deathnettles and combustilemons.

File: code/modules/antagonists/gang/gang.dm [hunk 1]
```diff
@@ -1,3 +1,7 @@
+#define INFLUENCE_OUTFIT 1
+#define INFLUENCE_TERRITORY 2
+#define INFLUENCE_SALARY 1
```
Comment (Archanial) [inline code/modules/antagonists/gang/gang.dm]: remember to undef this at the end

File: code/modules/antagonists/gang/gang.dm [hunk 2]
```diff
@@ -426,27 +431,17 @@
...
-				if(outfit.type in inner_outfits)
-					gang_outfit = outfit
-			if(gangster.wear_suit)
-				outfit = gangster.wear_suit
-				if(outfit.type in outer_outfits)
-					gang_outfit = outfit
-
-			if(gang_outfit)
+			if(check_gangster_outfit(gangmind.current))
 				gangster << "<span class='notice'>The [src] Gang's influence grows as you wear [gang_outfit].</span>"
```
Comment (Archanial) [inline code/modules/antagonists/gang/gang.dm]: please replace this this with proper function

File: code/modules/antagonists/gang/gang.dm [hunk 3]
```diff
@@ -426,27 +431,17 @@
...
 				uniformed++
 	return uniformed
 
+/datum/team/gang/proc/check_gangster_outfit(var/mob/living/carbon/human/gangster)
+	//Gangster must be alive and on station
+	if((gangster.stat == DEAD) || (is_station_level(gangster.z)))
```
Comment (Archanial) [inline code/modules/antagonists/gang/gang.dm]:
```suggestion
	if((gangster.stat == DEAD) || !(is_station_level(gangster.z)))
```
he must be on the station

**Comment by DatBoiTim** [comment]
> > Solid idea. Maybe take a pointer from goon with its gang locker as a way to make more gang points for gear. Inserting drugs into the locker gives points, but controlling territory gives a more steady income. So if a gang controls botany or chemistry they will be doing drugs? Gang with most points wins instead of territory as that is gonna end with people marking and walling off places with R walls to hide gang signs.
> 
> Hmm. I like the idea of winning based on influence.
> 
> But I don't like the idea of selling drugs for points. I was gonna have them feed money to the gang tool to rack up points and have to sell stuff for money.
> 
> Botany already has a lot of stuff that can help gangs, such as deathnettles and combustilemons.

Well you could make it take specific types of reagents in the container for points. Not to mention, if the drugs are worth enough, the combustlemon and nettles may not be as prevalent to make room for drug plants.

**Comment by LivingBatteryBox** [comment]
If you can only use tools while costumed, just keep costume in backpack and switch between it and normal clothes. Basically just becomes an inventory tax instead of actually incentivizing non-stealth tactics, at least for those who don't care about the income and just go for taking territory. I am absolutely a fan of the idea of gangs relying mostly on improvised weaponry and utility items, so I am really excited to see what this PR has in store.

**Comment by eeSPee** [comment]
> > > Solid idea. Maybe take a pointer from goon with its gang locker as a way to make more gang points for gear. Inserting drugs into the locker gives points, but controlling territory gives a more steady income. So if a gang controls botany or chemistry they will be doing drugs? Gang with most points wins instead of territory as that is gonna end with people marking and walling off places with R walls to hide gang signs.
> > 
> > 
> > Hmm. I like the idea of winning based on influence.
> > But I don't like the idea of selling drugs for points. I was gonna have them feed money to the gang tool to rack up points and have to sell stuff for money.
> > Botany already has a lot of stuff that can help gangs, such as deathnettles and combustilemons.
> 
> Well you could make it take specific types of reagents in the container for points. Not to mention, if the drugs are worth enough, the combustlemon and nettles may not be as prevalent to make room for drug plants.

Problem; This can lead to one gang getting key departments (biology, or whatever) and racking 100000 points, effectively snowballing and becoming unstoppable.

Leaves the other gangs having to catch up to 100 000 points, and gives the gang who did secure this separtment an endless supply of tools.

**Comment by ItsTisme** [comment]
It would be fun idea that gangs would actuely have conflict for "hot zones" but snowballing may be a problem. What if instead you get special grower machine that only works in specific area's randomly choosing as shift start. Gangs can fight for there terretories as they are more valueble. It would be ballanced by the fact that if they heavily guard 1 area then other gangs can cap other area's

**Comment by eeSPee** [comment]
Hmm. I think it's ready for review.

File: code/game/gamemodes/gangs/gang_items.dm [hunk 4]
```diff
@@ -269,112 +235,167 @@
...
+	cost = 200
+	item_path = /obj/item/storage/box/shuriken_box
 
-/datum/gang_item/weapon/frag
-	name = "Fragmentation Grenade"
-	id = "frag nade"
-	cost = 18
-	item_path = /obj/item/grenade/syndieminibomb/concussion/frag
+/obj/item/storage/box/shuriken_box
+	name = "Shuriken Box"
```
Comment (utilbird) [inline code/game/gamemodes/gangs/gang_items.dm]:
```suggestion
	name = "shuriken Box"
```

File: code/game/gamemodes/gangs/implant_gang.dm [hunk 5]
```diff
@@ -16,21 +16,23 @@
...
-				<b>Special Features:</b> This device will also emit a small EMP pulse, destroying any other implants within the host's brain.<BR>
+				<b>Special Features:</b> This device also contains healing nanites that can revive people already loyal to the organization.<BR>
 				<b>Integrity:</b> Implant's EMP function will destroy itself in the process."}
 	return dat
 
 /obj/item/implant/gang/implant(mob/living/target, mob/user, silent = 0)
 	if(!target || !target.mind || target.stat == DEAD)
 		return 0
+	if (HAS_TRAIT(target, TRAIT_MINDSHIELD))
+		target.visible_message("<span class='warning'>[target] seems to resist the implant!</span>", "<span class='warning'>You refuse the healing nanites from a kind stranger!</span>")
```
Comment (utilbird) [inline code/game/gamemodes/gangs/implant_gang.dm]: The visible message is a bit ambiguous, albeit funny. Imo the message should either be somewhat deceiving or just outright tell them

File: code/modules/antagonists/_common/antag_spawner.dm [hunk 6]
```diff
@@ -275,3 +275,35 @@
 	veil_msg = "<span class='warning'>You sense an adorable presence lurking just beyond the veil...</span>"
 	demon_type = /mob/living/simple_animal/slaughter/laughter
 	antag_type = /datum/antagonist/slaughter/laughter
+
+///////////GANGSTER REINFORCEMENT SPAWNER
+
+/obj/item/antag_spawner/gangster
+	name = "Crook Spawner"
```
Comment (utilbird) [inline code/modules/antagonists/_common/antag_spawner.dm]:
```suggestion
	name = "crook spawner"
```

**Comment by eeSPee** [comment]
Alright; latest change should address that.

**Comment by utilbird** [comment]
You renamed the datum instead of the object 😆

**Comment by eeSPee** [comment]
Huh?

**Comment by utilbird** [comment]
my review in `gang_items.dm`

**Comment by eeSPee** [comment]
> my review in `gang_items.dm`

Yea I saw it now; fixed!

**Review by utilbird — APPROVED** [review]
Everything looks good 👍

**Pull request merged**